testing/asserts_test.ts

```diff
@@ -1039,3 +1039,18 @@ Deno.test("Assert Throws Async Parent Error", () => {
     "Fail!",
   );
 });
+
+Deno.test("Assert Throws Async promise rejected with custom Error", async () => {
+  class CustomError extends Error {}
+  class AnotherCustomError extends Error {}
+  await assertThrowsAsync(
+    () =>
+      assertThrowsAsync(
+        () => Promise.reject(new AnotherCustomError("failed")),
+        CustomError,
+        "fail",
+      ),
+    AssertionError,
+    'Expected error to be instance of "CustomError", but was "AnotherCustomError".',
```

Comment:
:+1:

Reply:
@kt3k (I know this is already merged, but...)

Is there a convention for grammar in error messages?

For example, things like:
- terminating punctuation vs no terminating punctuation:
  - `"File not found."`
  - `"File not found"`
- conventional verbs (i.e. _got_/_saw_/_was_/etc.)
  - `"Expected x but saw y"`
  - `"Expected x but got y"`
- etc.

Reply:
@jsejcksn I don't think we have an explicit rule about it. But I think we should have one. Suggestions are welcome